tests/GcsBackupTest.php

```diff
@@ -0,0 +1,903 @@
+            $this->sapiClient,
+            $this->storageClient,
+            (string) getenv('TEST_GCP_BUCKET'),
+            'backup/',
+            true,
+        );
+        $backup->backupNotifications();
+
+        $targetFile = $this->storageClient
+            ->bucket((string) getenv('TEST_GCP_BUCKET'))
+            ->object('backup/notifications.json')
+            ->downloadAsString();
+
+        $data = (string) json_encode(
+            (array) json_decode($targetFile, true),
+            JSON_PRETTY_PRINT,
+        );
+
+        $expectedData = <<<JSON
+[
+    {
+        "id": "%s",
+        "event": "job-succeeded",
+        "filters": [
+            {
+                "field": "branch.id",
+                "value": "%s",
+                "operator": "=="
+            },
+            {
+                "field": "job.component.id",
+                "value": "keboola.orchestrator",
+                "operator": "=="
+            }
+        ],
+        "recipient": {
+            "channel": "email",
+            "address": "oj@oj.cz"
+        }
+    }
+]
+JSON;
+
+        self::assertStringMatchesFormat($expectedData, $data);
+    }
+
+    public function testBackupSignedUrls(): void
+    {
+        $component = new Components($this->branchAwareClient);
+
+        $config = new Configuration();
+        $config->setComponentId('keboola.snowflake-transformation');
+        $config->setDescription('Test Configuration');
+        $config->setConfigurationId('sapi-php-test');
+        $config->setName('test-configuration');
+        $configData = $component->addConfiguration($config);
+        $config->setConfigurationId($configData['id']);
+
+        $row = new ConfigurationRow($config);
+        $row->setChangeDescription('Row 1');
+        $row->setConfiguration(
+            ['name' => 'test 1', 'backend' => 'docker', 'type' => 'r', 'queries' => ['foo']],
+        );
+        $component->addConfigurationRow($row);
+
+        $backup = new GcsBackup(
+            $this->sapiClient,
+            $this->storageClient,
+            (string) getenv('TEST_GCP_BUCKET'),
+            'backup/',
+            true,
+        );
+        $backup->backupConfigs(false);
+        $backup->backupSignedUrls();
+
+        $targetContents = $this->storageClient
+            ->bucket((string) getenv('TEST_GCP_BUCKET'))
+            ->object('backup/signedUrls.json')
+            ->downloadAsString();
+
+        /** @var array $targetData */
+        $targetData = json_decode($targetContents, true);
+
+        self::assertArrayHasKey('configurations.json', $targetData);
+        self::assertArrayHasKey('configurations', $targetData);
+        self::assertArrayHasKey('keboola.snowflake-transformation', $targetData['configurations']);
+        self::assertArrayHasKey(
+            'sapi-php-test.json',
+            $targetData['configurations']['keboola.snowflake-transformation'],
+        );
+        self::assertStringContainsString(
```

Comment:
Protoze tam je ta signed url a pri chybe by se to vypsalo do logu v textu vyjimky, upravil bych to tak, aby tam bylo

`self::assertTrue()` a vevnitr porovnavat jestli jeden string obsahuje druhej.

Reply:
1/ ok, [upraveno](https://github.com/keboola/php-kbc-project-backup/pull/27/commits/e776fb73d66f865a4cf7a2085fabae0994b368c8), ale ->
2/ a to něčemu vadí že by se vypsala ta URL? tak by si někdo stáhl obsah testovací konfigurace - to mi je asi celkem jedno